src/configobject.h

```diff
@@ -34,23 +35,32 @@
 
 class ConfigKey {
   public:
-    ConfigKey();
+    ConfigKey(); // is required for qMetaTypeConstructHelper()
+    ConfigKey(const ConfigKey& key);
     ConfigKey(const QString& g, const QString& i);
-    ConfigKey(const char* g, const char* i);
     static ConfigKey parseCommaSeparated(QString key);
 
     inline bool isNull() const {
         return group.isNull() && item.isNull();
     }
 
+    // comparison function for ConfigKeys. Used by a QHash in ControlObject
+    inline bool operator==(const ConfigKey& other) const {
```

Comment:
It's common practice to implement comparison operators as free functions with left- and right-hand-side parameters (lhs/rhs) of the same type. Just add "friend" and a second parameter here.